ui/src/app/components/apps-panel/utils.tsx

```diff
@@ -168,3 +174,35 @@ export const toUserEnvironmentStatusByAppType = (
 export const showAppsPanel = (config: ConfigResponse) => {
   return config.enableCromwellGKEApp || config.enableRStudioGKEApp;
 };
+
+export interface AppState {
+  appType: UIAppType;
+  active: boolean;
+}
+
+const getAppState = (
+  runtime: Runtime | null | undefined,
+  userApps: UserAppEnvironment[],
+  appType: UIAppType
+): AppState => {
+  return {
+    appType,
+    active:
+      appType === UIAppType.JUPYTER
+        ? isRuntimeActive(runtime?.status)
+        : isAppActive(findApp(userApps, appType)),
+  };
+};
+
+export const getAppsByDisplayGroup = (
+  runtime: Runtime,
+  userApps: UserAppEnvironment[],
+  appsToDisplay: UIAppType[]
+) => {
+  const getAppStateWithContext = fp.partial(getAppState, [runtime, userApps]);
+  return fp.flow(
+    fp.map(getAppStateWithContext),
+    fp.orderBy(['active'], ['asc']),
+    fp.partition((app: AppState) => app.active)
```

Comment:
@jmthibault79, consider this. Because of the sort followed by the partition, I know the return type is a list of two lists where each item differs by the boolean. This is due to my familiarity with the `partition` function, assuming it acts like it does in other languages. I don't see anything unexpected, so I wouldn't normally add a comment.

Reply:
I was going to add a comment, but based on what you said @dmohs, I will not add it. The comment I had added is below. I am happy to add it, if it is useful:

```
```Returns a two element array. The first element is an array of apps
that are in the active state. The second element is an array of apps
that are in the available state (not active).

Reply:
I can appreciate those concerns. That makes sense to be mindful of.

I have two options:

Option One:
```
// Returns an array of the following form:
//     Outer Array
//       First Item:  Array of AppDisplayStates with active=true
//       Second Item:  Array of AppDisplayStates with active=false
```

Option Two (More verbose option):
```
// This function works in three steps:
// 1) Convert list of UIAppTypes to a list of AppDisplayStates (UIAppType plus active/available status)
// 2) Sort the list of AppDisplayStates based on active status.
// 3) Partitions the list into a list of two lists.
//    The first list is the list of AppDisplayStates that are in the active state, and the second list are the AppDisplayStates that are in the available state.
// Returns an array of the following form:
//     Outer Array
//       First Item:  Array of AppDisplayStates with active=true
//       Second Item:  Array of AppDisplayStates with active=false
```

I am including Option One for now. Please let me know if you would prefer Option Two or another configuration.